draft-ietf-taps-transport-security.md

```diff
@@ -249,11 +251,63 @@ interface or address changes.
 
 - Server: the peer responsible for responding to a session initiation.
 
+# Security Features
+
+In this section, we enumerate Security Features exposed by protocols discussed in the
+remainder of this document. Protocol security properties that are unrelated to the API surface
+exposed by such protocols, such as client or server identity hiding, are not listed 
+here as features.
+
+- Forward-secure key establishment: Cryptographic key establishment with forward secure properties.
+
+- Cryptographic algorithm negotiation: Negotiate support of protocol algorithms, including: encryption, 
+hash, MAC (PRF), and digital signature algorithms.
+
+- Stateful and stateless cross-connection session resumption: Connection establishment without needing 
+to complete an entirely new handshake. 
+
+- Peer authentication (certificate, raw public key, pre-shared key, or EAP-based): Peer authentication using 
+select or protocol-specific mechanisms.
+
+- Mutual authentication: Connection establishment wherein both endpoints are authenticated.
+
+- Record (channel or datagram) confidentiality and integrity: Encryption and authentication of 
+application plaintext bytes sent between peers over a channel or in individual datagrams.
```

Comment:
This also seems baseline.

Reply:
Yeah, though I left it in since RTP has optional authentication for some packets. Perhaps an API might allow folks to send things unencrypted, or unauthenticated? I'm not sure.

Reply:
I think this is still useful to call out as a feature even if standard.

Reply:
Confidentiality and integrity are separable, surely? SRTP allows for integrity protected but not encrypted packets, with reasonable use cases, and I believe also for encrypted but not authenticated.

Reply:
I'm wondering how much we want to be able to turn the knob between "insecure" (e.g., HTTP) and "secure" (encrypted + authenticated). If we want fine-grained control over that knob, then I think this makes sense to keep as optional. What I think I was advocating for a few months ago was a separation between security context establishment and integrity, such that (for instance) TAPS could support the use case of an OOB manifest with integrity information that could be used to authenticate data retrieved via HTTP, but I'm skeptical of the forward utility of unauthenticated communication. That said, if we leave it out, someone will offer an example and then we'll probably just have to put it back in.

Agreed that confidentiality is optional for a lot of use cases.

Reply:
We're certainly going to want "insecure", "authenticated/integrity protected", and "confidential plus authenticated/integrity protected", since there are deployed protocols that use those.